### PR TITLE
Fix segmentation fault when accessing invalid PVR channel from favourites

### PR DESCRIPTION
**Pull request by naseef**
## Description
This is a draft PR. I am basically a beginner to kodi and cpp. I 'm not sure if this is the correct way to tackle the issue.

Steps to reproduce segfault

1. add some m3u playlist to IPTV Simple PVR client (  I used `https://iptv-org.github.io/iptv/countries/in.m3u`)
2. Add some channels to favourites. Accessing the channel from the favourites menu works well at this point.
3. A few days later, the playlist provider (in my case, `https://iptv-org.github.io/iptv/countries/in.m3u` ) removes the channel from their list. 
4. Now accessing that channel from favourite menu causes segfault

I'm not sure about the `CGUIDialogKaiToast` and message text in particular. Someone please guide me
## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by ksooo — CHANGES_REQUESTED** [review]
I would keep it rather simple:

```c++
const std::shared_ptr<CPVRChannelGroupMember> groupMember = CServiceBroker...
if (groupMember)
  pvrItem = 
 ```
 
 No toast, no log entry, no early return needed.

**Comment by naseef** [comment]
@ksooo Changes inoroporated.

Noob question. Is the logging/dialog  possible in another layer (like in PVR IPTV Simple Client )? How the user get to know about his favourite entry is no longer available?

**Comment by ksooo** [comment]
> How the user get to know about his favourite entry is no longer available?

 That's a general unsolved problem and I would not come up with special fixes for certain use cases. What is needed is a general concept for handling invalid favourites entries. But this is definitely not for this PR.

File: xbmc/pvr/guilib/PVRGUIActions.cpp [hunk 1]
```diff
@@ -1551,9 +1551,14 @@ namespace PVR
...
     CFileItemPtr pvrItem(item);
     if (URIUtils::IsPVRChannel(item->GetPath()) && !item->HasPVRChannelInfoTag())
-      pvrItem = std::make_shared<CFileItem>(
+    {
+      const std::shared_ptr<PVR::CPVRChannelGroupMember> groupMember =
           CServiceBroker::GetPVRManager().ChannelGroups()->GetChannelGroupMemberByPath(
-              item->GetPath()));
+              item->GetPath());
+      if (!groupMember)
+        return false;
```
Comment (ksooo) [inline xbmc/pvr/guilib/PVRGUIActions.cpp:1559]: No need for the early return.

if (groupMember)
 pvrItem =
 
 Nothing else needed.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/pvr/guilib/PVRGUIActions.cpp (hunk 1 above)
Comment (naseef) [reply xbmc/pvr/guilib/PVRGUIActions.cpp]: done

File: xbmc/pvr/guilib/PVRGUIActions.cpp [hunk 2]
```diff
@@ -1551,9 +1551,13 @@ namespace PVR
   {
     CFileItemPtr pvrItem(item);
     if (URIUtils::IsPVRChannel(item->GetPath()) && !item->HasPVRChannelInfoTag())
-      pvrItem = std::make_shared<CFileItem>(
+    {
+      const std::shared_ptr<PVR::CPVRChannelGroupMember> groupMember =
```
Comment (ksooo) [inline xbmc/pvr/guilib/PVRGUIActions.cpp:1555]: Please remove the `PVR::` namespace. It's not needed here.

**Review by ksooo — CHANGES_REQUESTED** [review]

**Review by ksooo — APPROVED** [review]
Thank you for your contribution.

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**